src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -77,6 +77,16 @@ def getModuleInfoForEnvironment(env)
     end
   end
 
+  def getAllModuleInfoForEnvironment()
+    all_envs = @env_loader.list
+    all_envs.collect do |env|
```

Comment:
That data structure does seem easier to work with...then hopefully the function I use to process it on the clojure side might not have to be so complicated.